libjolie/src/jolie/lang/parse/ast/types/TypeChoiceDefinition.java

```diff
@@ -0,0 +1,96 @@
+package jolie.lang.parse.ast.types;
+
+import jolie.lang.NativeType;
+import jolie.lang.parse.OLVisitor;
+import jolie.lang.parse.context.ParsingContext;
+import jolie.util.Range;
+
+import java.util.Map;
+import java.util.Set;
+
+public class TypeChoiceDefinition extends TypeDefinition {
+    private final TypeDefinition T1;
```

Comment:
Variables should not start with an upper case. I propose "left", and "right" instead of T2.